### PR TITLE
KTOR-6501 Rewrite KDocs for HttpClient.close, HttpStatement, and Http…

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt [hunk 1]
```diff
@@ -7,12 +7,33 @@ package io.ktor.client.request
...
+ * val response = client.request {
+ *     url("https://ktor.io")
+ *     method = HttpMethod.Get
+ * }
+ * ```
+ * This example demonstrates making a GET request to "https://ktor.io".
+ *
+ * ## Note
+ * In addition to this generic `request` method, there are more specific extension functions, such as
+ * [HttpClient.get], [HttpClient.post], HttpClient.put`, and `HttpClient.delete`, which are often more
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt:29]:
```suggestion
 * [HttpClient.get], [HttpClient.post], [HttpClient.put], and [HttpClient.delete], which are often more
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt [hunk 2]
```diff
@@ -26,9 +47,31 @@ public suspend inline fun HttpClient.prepareRequest(
...
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt:67]: Same as above.

File: ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt [hunk 3]
```diff
@@ -40,9 +83,30 @@ public suspend inline fun HttpClient.prepareRequest(block: HttpRequestBuilder.()
...
+ * val client = HttpClient()
+ * val response = client.request("https://ktor.io") {
+ *     method = HttpMethod.Get
+ * }
+ * ```
+ * This example demonstrates making a GET request to "https://ktor.io".
+ *
+ * ## Note
+ * In addition to this generic `request` method, there are more specific extension functions, such as
+ * [HttpClient.get], [HttpClient.post], HttpClient.put`, and `HttpClient.delete`, which are often more
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt:102]: Same here.

File: ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt [hunk 4]
```diff
@@ -64,9 +128,30 @@ public suspend inline fun HttpClient.prepareRequest(
...
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/request/builders.kt:147]: And here.

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpStatement.kt [hunk 5]
```diff
@@ -6,21 +6,22 @@ package io.ktor.client.statement
...
 import io.ktor.utils.io.*
-import io.ktor.utils.io.charsets.*
-import io.ktor.utils.io.core.*
 import kotlinx.coroutines.*
 
 /**
- * Prepared statement for a HTTP client request.
- * This statement doesn't perform any network requests until [execute] method call.
- * [HttpStatement] is safe to execute multiple times.
+ * Represents a prepared HTTP request statement for `HttpClient`.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpStatement.kt:15]:
```suggestion
 * Represents a prepared HTTP request statement for [HttpClient].
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpStatement.kt [hunk 6]
```diff
@@ -6,21 +6,22 @@ package io.ktor.client.statement
...
+ *
+ * The `HttpStatement` class encapsulates a request configuration without executing it immediately.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpStatement.kt:17]:
```suggestion
 * The [HttpStatement] class encapsulates a request configuration without executing it immediately.
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/utils/CoroutineUtils.kt [hunk 7]
```diff
@@ -8,7 +8,7 @@ import io.ktor.utils.io.*
 import kotlinx.coroutines.*
 
 /**
- * Creates [CoroutineDispatcher] for client with fixed [threadCount] and specified [dispatcherName].
+ * Creates [CoroutineDispatcher] for the client with fixed [threadCount] and specified [dispatcherName].
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/utils/CoroutineUtils.kt:11]: Here we can also use `@param` to describe `threadCount` and `dispatcherName` in the same way it is done in the other KDocs.

**Review by vnikolova — CHANGES_REQUESTED** [review]
I noticed in some places elements weren't linked (see in comments, but there might be more). Otherwise looks great!

**Comment by e5l** [comment]
Hey @vnikolova, thanks for the review. Let me check the comments!

**Comment by e5l** [comment]
@vnikolova, fixed!

**Review by vnikolova — APPROVED** [review]

**Pull request merged**